Automattic/pocket-casts-ios
# End of Year: Add some common views and improve centering for larger sizes

| 📘 Project: #374 | 🛫 Depends on: #562 |
|:---:|:---:|

This includes some common views to use throughout the stories:
- `StoryLabelContainer`
   - This wraps the title and subtitle labels to enforce consistent spacing and top padding
   - This also uses a dynamic scale for the label spacing and top padding to allow the label spacing to reduce on smaller height devices
- `PodcastStackView`
  - This renders the 3 podcast stacks 
- `PodcastCoverContainer`
  - This wraps most of the story views and applies consistent top and bottom spacing across stories
  - This also introduces using a slightly larger top padding for larger devices to ensure they look more balanced and centered


## Screenshots

### Story 3: Before and After Comparisons
| iPhone SE - Before | After |
|:---:|:---:|
|<img src="https://user-images.githubusercontent.com/793774/205213633-042fbe4a-f89d-49cc-816a-5f439ccf5980.png" width="320" />|<img src="https://user-images.githubusercontent.com/793774/205349294-742bebaa-6118-44f8-af57-38457bd3f37e.png" width="320" />|

| iPhone 13 - Before | After |
|:---:|:---:|
|<img src="https://user-images.githubusercontent.com/793774/205350164-b8e6b0ca-2ac3-4c73-acd2-80c113431dcc.png" width="320" />|<img src="https://user-images.githubusercontent.com/793774/205349291-d0738840-1b5e-40fb-bebf-3f319a47869e.png" width="320" />|

| iPhone 14 Pro Max - Before | After |
|:---:|:---:|
|<img src="https://user-images.githubusercontent.com/793774/205350359-631fd251-f51d-420b-b90d-a01e9d0e8fb8.png" width="320" />|<img src="https://user-images.githubusercontent.com/793774/205349289-8b824535-2fef-451f-8952-fa3fddd02417.png" width="320" />|

### Story 5: Before and After Comparisons
| iPhone SE - Before | After |
|:---:|:---:|
|<img src="https://user-images.githubusercontent.com/793774/205229714-bb38735c-70f1-450c-a049-3ab3d7b675e8.png" width="320" />|<img src="https://user-images.githubusercontent.com/793774/205344579-1b7e02c2-7f03-4616-a856-baa2c0b5cfc7.png" width="320" />|

| iPhone 13 - Before | After |
|:---:|:---:|
|<img src="https://user-images.githubusercontent.com/793774/205229715-0113c896-303e-4fcd-9651-070c92dbe439.png" width="320" />|<img src="https://user-images.githubusercontent.com/793774/205344576-49b98a7c-086e-4403-adfb-5e6de39557c6.png" width="320" />|

| iPhone 14 Pro Max - Before | After |
|:---:|:---:|
|<img src="https://user-images.githubusercontent.com/793774/205229711-c248d155-c508-4877-928b-c418fa5c20d6.png" width="320" />|<img src="https://user-images.githubusercontent.com/793774/205345609-aa9f1ddc-f08a-410b-adf5-c5c5ec2014db.png" width="320" />|

## To test

1. Launch the app
2. Tap the Profile Tab > End of Year card
3. Cycle through the stories
4. ✅ Verify the 3rd, 5th, 6th, and 8th story content appears relatively centered 
5. ✅ Repeat the steps on various device sizes
6. ✅ On smaller screens the space between the title and subtitle labels is reduced and looks good. 

## Checklist
